soc/riscv/riscv-privilege/nuclei/gd32vf103/Kconfig.defconfig.gd32vf103xxxx

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2021 Tokita, Hiroshi <tokita.hiroshi@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_GD32VF103C4T6
+
+config SOC
+	default "gd32vf103c4t6"
+
```

Comment:
there is no need to introduce Kconfig settings for all SoC variants, better follow ARM SoC approach (`SOC_GD32VF103`)

ref
https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/gigadevice/gd32f4xx/Kconfig.soc
https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/gigadevice/gd32f4xx/Kconfig.defconfig.gd32f450

Reply:
Unify all variants into gd32vf103.